From 90ae505c06fddcc08622594bb82bb449511c7e9a Mon Sep 17 00:00:00 2001
From: Matthew Fatheree <Matthew.Fatheree@belkin.com>
Date: Sat, 19 Apr 2014 00:08:43 +0700
Subject: [PATCH 07/34] mamba mvebu: force nand driver to use 4-bit ECC

if the NAND flash's recommended ECC strength is less than 4-bit, for to
use 4-bit ECC.

Signed-off-by: Matthew Fatheree <Matthew.Fatheree@belkin.com>
---
 .../0306-ARM-mvebu-mamba-nand-force-4bit-ecc.patch |   14 ++++++++++++++
 1 file changed, 14 insertions(+)
 create mode 100644 target/linux/mvebu/patches-3.10/0306-ARM-mvebu-mamba-nand-force-4bit-ecc.patch

diff --git a/target/linux/mvebu/patches-3.10/0306-ARM-mvebu-mamba-nand-force-4bit-ecc.patch b/target/linux/mvebu/patches-3.10/0306-ARM-mvebu-mamba-nand-force-4bit-ecc.patch
new file mode 100644
index 0000000..111aff5
--- /dev/null
+++ b/target/linux/mvebu/patches-3.10/0306-ARM-mvebu-mamba-nand-force-4bit-ecc.patch
@@ -0,0 +1,14 @@
+--- a/drivers/mtd/nand/pxa3xx_nand.c
++++ b/drivers/mtd/nand/pxa3xx_nand.c
+@@ -1517,6 +1517,11 @@ KEEP_CONFIG:
+ 		}
+ 	}
+ 
++#ifdef CONFIG_MACH_ARMADA_MAMBA
++	/* force to use 4-bit ECC */
++	if (chip->ecc_strength_ds < 4)
++		chip->ecc_strength_ds = 4;
++#endif
+ 	ecc_strength = chip->ecc_strength_ds;
+ 	ecc_step = chip->ecc_step_ds;
+ 
-- 
1.7.9.5

